### ISSUE TITLE
Client v4.0.0 TransactionError parsing not correct

### ISSUE THREAD

**Issue by openinformation** [recurly/recurly-client-python#482]
**Describe the bug**

With client 4.0.0, 3DS flow handling as described in the docs does not work and is only possible as a workaround. 

The error cast by the client has `transaction_error` set to `None` by `Resource.cast_json`, and it would be only possible to access the transaction error's details via the original response, which is still stored with the resource.

**To Reproduce**

Steps to reproduce the behavior.

1.  trigger a Recurly billing info update via `recurly.js`, using test card `4000000000003220` to trigger 3DS flow
2. get the recurly token, e.g. "7Mw0MqrarDbEyNFNmSaJYA"
3. do `client.update_billing_info(account_id, {"token_id": "7Mw0MqrarDbEyNFNmSaJYA"})
4. client correctly raises `recurly.errors.TransactionError` as described in the docs
5. but the error has `transaction_error` set to `None`, although docs say this is where the required 3DS token is stored
6. the raw repsonse by Recurly is correct, however:

```
{'error': {'type': 'transaction', 'message': 'Your card must be authenticated with 3-D Secure before continuing.', 'params': [{'param': 'Billing info', 'message': 'Your card must be authenticated with 3-D Secure before continuing.'}], 'transaction_error': {'object': 'transaction_error', 'transaction_id': 'ok1xhh2dz0a9', 'category': 'three_d_secure_action_required', 'code': 'three_d_secure_action_required', 'message': 'Your card must be authenticated with 3-D Secure before continuing.', 'merchant_advice': 'Your payment gateway is requesting that the transaction be completed with 3-D Secure.', 'three_d_secure_action_token_id': 'STZUxb_QYbpzBydwROMf3Q'}}}
```

**Expected behavior**

`recurly.errors.TransactionError` must include the parsed `transaction_error`, as described here: https://developers.recurly.com/tag/billing_info#operation/update_a_billing_info

**Your Environment**

- Which version of this library are you using?
  - `Recurly 4.0.0`
- Which version of the language are you using?
  - `Python 3.7.5`

**Comment by openinformation** [comment]
Looking at the latest changes to master, it seems that the transaction error handling is fixed/re-introduced from older branch? https://github.com/recurly/recurly-client-python/pull/479

So is an updated `4.0.1` version around the corner, or are there more changes in the pipeline? Just to get a rough idea of an ETA...

**Comment by douglasmiller** [comment]
@openinformation, yes the `4.0.1` patch was published a few hours ago. Please let us know if you're still having issues. Sorry about the confusion

**Comment by openinformation** [comment]
@douglasmiller will continue the switch to v4 in 2 weeks and give feedback then

**Comment by openinformation** [comment]
@douglasmiller can confirm that issue is solved with `v4.1.0`

**Comment by douglasmiller** [comment]
@openinformation, yes this should be fixed as of `4.0.1`. The fix for using `ErrorMayHaveTransaction` is [present on the `4.1.0` release](https://github.com/recurly/recurly-client-python/blob/4.1.0/recurly/resource.py#L38) as well.

**Comment by douglasmiller** [comment]
I'm going to close this issue, but please don't hesitate to let us know if you have any other problems.
